clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/DcosClientProvider.groovy

```diff
@@ -1,27 +1,32 @@
 package com.netflix.spinnaker.clouddriver.dcos
 
-import com.netflix.spectator.api.Registry
-
-import java.util.concurrent.ConcurrentHashMap
-
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import mesosphere.dcos.client.DCOS
 import mesosphere.dcos.client.DCOSClient
 
+import java.util.concurrent.ConcurrentHashMap
+
 class DcosClientProvider {
 
   private final Map<String, DCOS> dcosClients = new ConcurrentHashMap<>()
-  private final Registry registry
+  private final AccountCredentialsProvider credentialsProvider
 
-  DcosClientProvider(Registry registry) {
-    this.registry = registry
+  DcosClientProvider(AccountCredentialsProvider credentialsProvider) {
+    this.credentialsProvider = credentialsProvider
   }
 
-  DCOS getDcosClient(DcosCredentials account) {
-    String key = account.name
-    return dcosClients.computeIfAbsent(key, { k -> DCOSClient.getInstance(account.dcosUrl, account.dcosClientConfig) })
+  DCOS getDcosClient(DcosCredentials credentials) {
+    String key = credentials.name
+    return dcosClients.computeIfAbsent(key, { k -> DCOSClient.getInstance(credentials.dcosUrl, credentials.dcosClientConfig) })
   }
 
   DCOS getDcosClient(String accountName) {
```

Comment:
I think we added this overload recently - didn't catch this in the review but it wouldn't work as it was, since it would depend on the other overload being called first to populate the dcosClients cache for that account key.